src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,73 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    bool rc = false;
+    auto transpose = as_type_ptr<op::Transpose>(n);
+
+    if (transpose)
+    {
+        auto perm = as_type_ptr<op::Constant>(n->get_argument(1));
+        auto perm_value = perm->get_vector<int64_t>();
+        auto data = n->get_argument(0);
+        auto data_shape = n->input_value(0).get_partial_shape().to_shape();
+
+        // Data shape should be >= 3
+        if (data_shape.size() < 3 || data_shape.size() > 4)
```

Comment:
Why do you have such a limitations? What is the reason do not support  2D and 5D+ tensors?

Reply:
because we dont deal with 5D tensor and @wenzhe-nrv had a optimization to remove out useless transpose for 2D vectors, hence I had it as < 3 and > 4..
anyway, removed it and now its generic and can work for n dims